Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1

```diff
@@ -372,12 +372,80 @@ function Publish-UpgradedServiceFabricApplication
         }
 
         Write-Host (Get-VstsLocString -Key SFSDK_WaitingForUpgrade)
-        $upgradeStatusFetcher = { Get-ServiceFabricApplicationUpgradeAction -ApplicationName $ApplicationName }
+        $unhealthyEvaluationMessageFunction = {
+            param(
+                $UnhealthyEvaluations,
+
+                [string]$Indentation
+            )
+
+            if ($UnhealthyEvaluations -eq $null) {
+                return ""
+            }
+
+            $indentatedErrorString = ""
+            foreach ($UnhealthyEvaluation in $UnhealthyEvaluations) {
+                $indentatedErrorString += $Indentation + $UnhealthyEvaluation.Description
+                # see if indentation needs to be increased. based on the type of evaluation.
+                $indentatedErrorString += & $unhealthyEvaluationMessageFunction $UnhealthyEvaluation, if($UnhealthyEvaluations.kind -eq ($UnhealthyEvaluation.kind + "s")) { return $Indentation } Else { return $Indentation + "`t" }
+            }
+
+            if ($UnhealthyEvaluation.UnhealthyEvent) {
```

Comment:
which level contains UnhealthyEvent? is it only leaf level or can be at any depth?

Reply:
I have only seen it to be in the deepest level only.